yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -66,6 +66,8 @@ private[spark] class Client(
   private val executorMemoryOverhead = args.executorMemoryOverhead // MB
   private val distCacheMgr = new ClientDistributedCacheManager()
   private val isClusterMode = args.isClusterMode
+  private val fireAndForget = isClusterMode &&
+    sparkConf.getBoolean("spark.yarn.cluster.quiet", false)
```

Comment:
when I read cluster.quiet I think more of logging and not application submission.  I would rather see this config called something like spark.yarn.waitForCompletion.  it feels like we should have an option to spark-submit for this but perhaps for now leave it a config and then if people want it add that later. 

You also need to document the config in the .md file for yarn.